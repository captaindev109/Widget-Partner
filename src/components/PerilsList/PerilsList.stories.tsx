import React from 'react';
import { Meta, Story } from '@storybook/react';
import styled from 'styled-components';
import { decorators } from '@/../.storybook/preview';
import Wrapper from '@/generics/Wrapper';
import { PerilsList, PerilsListProps, Peril } from './PerilsList';
import { PERILS_MOCK } from '@/mocks/quoteCart_data';

// A wrapper to position the component
const StoryWrapper = styled(Wrapper)`
  width: 100%;
`;

// Name the folder to either generics or components
const meta: Meta = {
  title: 'Components/PerilsList',
  component: PerilsList,
  decorators,
};

export default meta;

// Showcasing the component
const Template: Story<PerilsListProps> = () => {
  return (
    <StoryWrapper>
      <PerilsList perils={PERILS_MOCK} />
    </StoryWrapper>
  );
};

export const DefaultPerilsList = Template.bind({});

DefaultPerilsList.args = {};

// Showcasing the component
const PerilTemplate: Story<PerilsListProps> = () => {
  const peril = PERILS_MOCK[0];
  return (
    <StoryWrapper>
      <Peril index={0} onClick={() => null} peril={peril} />
    </StoryWrapper>
  );
};

export const DefaultPeril = PerilTemplate.bind({});

DefaultPeril.args = {};
