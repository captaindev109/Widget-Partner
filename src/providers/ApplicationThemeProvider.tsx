import { ThemeProvider, DefaultTheme } from 'styled-components';
import GlobalStyles from '@/styles/Global';
import React from 'react';
import { ChildrenReturnType } from '@/types/types';

interface ApplicationThemeProviderProps {
  children?: ChildrenReturnType;
  theme: DefaultTheme;
}

const ApplicationThemeProvider = ({ children, theme }: ApplicationThemeProviderProps) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyles />
      {children}
    </ThemeProvider>
  );
};

export default ApplicationThemeProvider;
