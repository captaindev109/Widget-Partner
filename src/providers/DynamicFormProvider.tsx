import React from 'react';
import { ChildrenReturnType } from '@/types/types';
import { FormProvider, useForm } from 'react-hook-form';

interface DynamicFormProviderProps {
  children?: ChildrenReturnType;
}

const useDynamicRootForm = () => {
  const methods = useForm({ mode: 'onChange' });

  return { methods };
};

export const DynamicFormProvider = ({ children }: DynamicFormProviderProps) => {
  const { methods } = useDynamicRootForm();

  return <FormProvider {...methods}>{children}</FormProvider>;
};
