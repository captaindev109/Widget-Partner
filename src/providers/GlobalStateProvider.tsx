import { mainMachine } from '@/machines/courseMachine';
import headerMachine from '@/machines/headerMachine';
import footerMachine from '@/machines/footerMachine';
import React, { createContext } from 'react';
import { useInterpret } from '@xstate/react';
import { ActorRefFrom } from 'xstate';
import { storeMachine } from '@/machines/storeMachine';

interface GlobalStateContextProps {
  mainService: ActorRefFrom<typeof mainMachine>;
  headerService: ActorRefFrom<typeof headerMachine>;
  footerService: ActorRefFrom<typeof footerMachine>;
  storeService: ActorRefFrom<typeof storeMachine>;
}

export const GlobalStateContext = createContext({} as GlobalStateContextProps);

interface GlobalStateProviderProps {
  children?: React.ReactNode;
}

const useDevTool = process.env.NODE_ENV === 'development';

export const GlobalStateProvider = ({ children }: GlobalStateProviderProps) => {
  const services = {
    mainService: useInterpret(mainMachine, {
      devTools: useDevTool,
    }),
    headerService: useInterpret(headerMachine, {
      devTools: useDevTool,
    }),
    footerService: useInterpret(footerMachine, {
      devTools: useDevTool,
    }),
    storeService: useInterpret(storeMachine, {
      devTools: useDevTool,
    }),
  };

  return <GlobalStateContext.Provider value={services}>{children}</GlobalStateContext.Provider>;
};
