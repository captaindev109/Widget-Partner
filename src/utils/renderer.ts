import React from 'react';
import { ConfigProps, Index, EachComponentObjectChildren, BasicAny } from '@/types/types';
import { UIComponents } from './components';

const recursivelyCreateReactComponents = (
  componentToCreate: BasicAny,
  configuration: ConfigProps,
  children: BasicAny
) => {
  if (typeof componentToCreate !== 'undefined') {
    return React.createElement(
      componentToCreate,
      configuration,
      children &&
        (typeof children === 'string'
          ? children
          : children.map((component: EachComponentObjectChildren, index: Index) => viewRenderer(component, index)))
    );
  }
  null;
};

const viewRenderer: BasicAny = (config: ConfigProps, index: Index) => {
  const children = config.children;
  const newConfig = { ...config, key: index };
  const LookForComponentInUILibrary = UIComponents[config.component];
  return recursivelyCreateReactComponents(LookForComponentInUILibrary, newConfig, children);
};

export default viewRenderer;
