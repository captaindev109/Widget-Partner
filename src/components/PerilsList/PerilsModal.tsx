import Card from '@/generics/Card';
import Image from '@/generics/Image';
import Text from '@/generics/Text';
import { PerilV2 } from '@/types/schemaTypes';
import { ANIMATION_DURATION } from '@/utils/animationUtils';
import { device } from '@/utils/constants';
import { forwardRef, ForwardRefRenderFunction } from 'react';
import styled from 'styled-components';
import { List, ListItem } from '../List';
import Modal, { ModalRef } from '../Modal/Modal';
import { CreateSVGComponent } from '../SVGIcons/Index';

// TODO - Add internationalization

export interface PerilsModalProps {
  perils: Array<PerilV2>;
  selectedPeril: PerilV2 | null;
  setSelectedPeril: (peril: PerilV2) => void;
  ref: ModalRef;
}

const ModalContentWrapper = styled(Card)`
  display: flex;
  flex-direction: column;

  width: 90vw;
  height: 90vh;

  @media ${device.tablet} {
    width: 70vw;
    height: 70vh;
  }
`;

const PerilSlideList = styled(List)`
  display: flex;
  width: 100%;
  overflow-x: scroll;

  box-shadow: 0px 4px 12px rgba(0, 0, 0, 0.05);
`;

const PerilSliderContent = styled(Card)<{ selected: boolean }>`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  cursor: pointer;

  &:hover {
    border-color: ${(props) => !props.selected && props.theme.color.gray500};
  }

  padding: ${(props) => props.theme.size.padding.p8};

  transition: border ${ANIMATION_DURATION} ease-in-out;

  border-bottom: 2px solid ${(props) => (props.selected ? props.theme.color.gray900 : props.theme.color.gray300)};

  height: 100px;
  width: 100px;
`;

const PerilTextWrapper = styled(Card)`
  display: flex;
  width: 100%;
  justify-content: center;
  align-items: center;
`;

const PerilSliderText = styled(Text)`
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`;

const InformationWrapper = styled(Card)`
  display: flex;
  flex: 1;
  flex-direction: column;
  padding: ${(props) => props.theme.size.padding.p32};
  overflow-y: scroll;

  @media ${device.tablet} {
    padding: ${(props) => props.theme.size.padding.p56};
  }
`;

const Headline = styled(Text)`
  margin-bottom: ${({ theme }) => theme.size.margin.m16};
`;

const HeadlineWrapper = styled(Card)`
  margin-bottom: ${({ theme }) => theme.size.margin.m20};
`;

const IncludedWrapper = styled(Card)`
  display: flex;
  flex-direction: column;
  margin-top: ${({ theme }) => theme.size.margin.m16};

  @media ${device.tablet} {
    & > *:first-child {
      padding-right: ${({ theme }) => theme.size.padding.p16};
    }
    & > *:last-child {
      padding-left: ${({ theme }) => theme.size.padding.p16};
    }
    flex-direction: row;
  }
`;

const IncludedContentWrapper = styled(Card)`
  display: flex;
  flex-direction: column;
  flex: 1;
  margin-bottom: ${({ theme }) => theme.size.margin.m20};
`;

const InfoWrapper = styled(Card)`
  position: relative;
  padding: 24px 40px;
  background: ${({ theme }) => theme.color.gray100};
  border-radius: ${({ theme }) => theme.size.borderRadius.br4};
`;

const IncludedTitle = styled(Text)`
  margin-bottom: ${({ theme }) => theme.size.margin.m8};
`;

const InfoIcon = styled(Card)`
  position: absolute;
  left: 12px;
`;

const IconTextWrapper = styled(Card)`
  display: flex;
  flex-direction: row;
  word-break: break-word;
  margin-bottom: ${({ theme }) => theme.size.margin.m8};
`;

const Icon = styled(Card)`
  display: flex;
  height: ${({ theme }) => theme.size.lineHeights.lh4};
  flex-direction: row;
  align-items: center;
  margin-right: ${({ theme }) => theme.size.margin.m8};
`;

const PerilsModal: ForwardRefRenderFunction<ModalRef, PerilsModalProps> = (
  { perils, selectedPeril, setSelectedPeril },
  ref
) => {
  return (
    <Modal ref={ref}>
      <ModalContentWrapper>
        <PerilSlideList>
          {perils.map((peril: PerilV2) => (
            <ListItem key={peril.title}>
              <PerilSliderContent
                onClick={() => setSelectedPeril(peril)}
                selected={selectedPeril?.title === peril.title}
              >
                <Image alt={peril.title} src={peril.icon.variants.light.svgUrl} width="40px" height="40px" />
                <PerilTextWrapper>
                  <PerilSliderText variant="Body3">{peril.title}</PerilSliderText>
                </PerilTextWrapper>
              </PerilSliderContent>
            </ListItem>
          ))}
        </PerilSlideList>
        <InformationWrapper>
          <HeadlineWrapper>
            <Headline variant="HeadlineM">{selectedPeril?.title}</Headline>
            <Text variant="Body1">{selectedPeril?.description}</Text>
          </HeadlineWrapper>
          <IncludedWrapper>
            {selectedPeril?.covered && (
              <IncludedContentWrapper>
                <IncludedTitle weight="bold" variant="Body2">
                  Det h채r t채cks
                </IncludedTitle>
                {selectedPeril?.covered.map((item) => (
                  <IconTextWrapper key={item}>
                    <Icon>{CreateSVGComponent('Covered')}</Icon>
                    <Text variant="Body2">{item}</Text>
                  </IconTextWrapper>
                ))}
              </IncludedContentWrapper>
            )}
            {!!selectedPeril?.exceptions.length && (
              <IncludedContentWrapper>
                <IncludedTitle weight="bold" variant="Body2">
                  Undantag
                </IncludedTitle>
                {selectedPeril?.exceptions.map((item) => (
                  <IconTextWrapper key={item}>
                    <Icon>{CreateSVGComponent('Exception')}</Icon>
                    <Text variant="Body2">{item}</Text>
                  </IconTextWrapper>
                ))}
              </IncludedContentWrapper>
            )}
          </IncludedWrapper>
          {!!selectedPeril?.info && (
            <InfoWrapper>
              <InfoIcon>{CreateSVGComponent('Information')}</InfoIcon>
              <Text weight="bold" variant="Body2">
                Att t채nka p책
              </Text>
              <Text variant="Body3">{selectedPeril?.info}</Text>
            </InfoWrapper>
          )}
        </InformationWrapper>
      </ModalContentWrapper>
    </Modal>
  );
};

export default forwardRef(PerilsModal);
