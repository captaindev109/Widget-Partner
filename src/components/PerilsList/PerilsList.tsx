import Card from '@/generics/Card';
import Image from '@/generics/Image';
import Text from '@/generics/Text';
import { PerilV2 } from '@/types/schemaTypes';
import { BasicAny } from '@/types/types';
import { device } from '@/utils/constants';
import { FC, useRef, useState } from 'react';
import styled from 'styled-components';
import { List, ListItem } from '../List';
import { ModalRef } from '../Modal/Modal';
import PerilsModal from './PerilsModal';

export interface PerilsListProps {
  perils: Array<PerilV2>;
}

export interface PerilProps {
  onClick: (event: BasicAny) => BasicAny;
  peril: PerilV2;
  index: number;
}

const PerilsListItem = styled(ListItem)`
  @media ${device.mobileS} {
    width: 50%;
  }
  @media ${device.tablet} {
    width: 25%;
  }
`;

const StyledPeril = styled(Card)<{ index: number }>`
  display: flex;
  align-items: center;
  flex-direction: row;
  cursor: pointer;

  transition: box-shadow 0.1s ease-in-out;

  &:hover {
    box-shadow: 0px 4px 12px rgba(0, 0, 0, 0.05);
  }

  margin-top: ${(props) => props.theme.size.margin.m4};
  margin-bottom: ${(props) => props.theme.size.margin.m4};

  @media ${device.mobileS} {
    padding: 4px 8px;
    height: 56px;
    ${(props) => props.index % 2 === 0 && `margin-right: ${props.theme.size.margin.m4}`};
    ${(props) => props.index % 2 === 1 && `margin-left: ${props.theme.size.margin.m4}`};

    & > p {
      padding-left: ${({ theme }) => theme.size.padding.p8};
    }
  }

  @media ${device.tablet} {
    flex-direction: column;
    justify-content: space-between;
    align-items: flex-start;
    height: 140px;

    & > img {
      width: 48px;
      height: 48px;
    }

    & > p {
      padding-left: 0px;
    }

    padding: ${(props) => props.theme.size.padding.p20};
    margin-left: ${(props) => (props.index % 4 === 0 ? 0 : props.theme.size.margin.m10)};
    margin-right: ${(props) => (props.index % 4 === 3 ? 0 : props.theme.size.margin.m10)};
  }

  background-color: ${({ theme }) => theme.color.white};
  border-radius: ${({ theme }) => theme.size.borderRadius.br4};
`;

const PerilText = styled(Text)`
  width: 100%;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`;

//The main component with dynamic rendering and state
export const Peril: FC<PerilProps> = ({ onClick, peril, index }) => {
  const handleClick = () => {
    onClick(peril);
  };

  return (
    <StyledPeril index={index} data-testid="Peril" shadow={1} onClick={handleClick}>
      <Image src={peril.icon.variants.light.svgUrl} alt={peril.title} width="20px" height="20px" />
      <PerilText variant="Body2">{peril.title}</PerilText>
    </StyledPeril>
  );
};

export const PerilsList: FC<PerilsListProps> = ({ perils }) => {
  const modalRef = useRef<ModalRef>(null);
  const [selectedPeril, setSelectedPeril] = useState<PerilV2 | null>(null);

  const handlePerilClick = (peril: PerilV2) => {
    setSelectedPeril(peril);
    modalRef.current?.toggle?.();
  };

  return (
    <>
      <List display="flex" direction="row" wrap="wrap">
        {perils.map((peril: PerilV2, index) => (
          <PerilsListItem key={peril.title}>
            <Peril data-testid={`Peril-${index}`} index={index} peril={peril} onClick={handlePerilClick} />
          </PerilsListItem>
        ))}
      </List>
      <PerilsModal ref={modalRef} perils={perils} selectedPeril={selectedPeril} setSelectedPeril={setSelectedPeril} />
    </>
  );
};

export default PerilsList;
