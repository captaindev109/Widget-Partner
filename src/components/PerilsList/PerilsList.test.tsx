/**
 *
 *  ********** Unit & Snapshot Testing *********
 *
 * Unit test is specifiic for each functionliaty of the component
 * Snapshot test is to track Over All changes in the code since the first snap
 **/

import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { GlobalStateProvider } from '@/providers/GlobalStateProvider';
import defaultTheme from '@/theme/theme';
import ApplicationThemeProvider from '@/providers/ApplicationThemeProvider';
import PerilsList from '.';
import { PERILS_MOCK } from '@/mocks/quoteCart_data';
import { Peril } from './PerilsList';

const MockComponent: React.FC = () => {
  return (
    <GlobalStateProvider>
      <ApplicationThemeProvider theme={defaultTheme}>
        <PerilsList perils={PERILS_MOCK} />
      </ApplicationThemeProvider>
    </GlobalStateProvider>
  );
};

/** Render tests */
describe('Peril render tests', () => {
  test('to see if the component render without crashing', () => {
    render(<MockComponent />);
  });
});

/** Snapshot test */
describe('Peril snapshot test', () => {
  it('Snapshot test should pass', () => {
    const snap = renderer.create(<MockComponent />).toJSON();
    expect(snap).toMatchSnapshot();
  });
});

/** Peril click tests */
describe('Peril click test', () => {
  it('Test the Peril flow', () => {
    const peril = PERILS_MOCK[0];
    const handleClick = jest.fn();
    const { getByTestId } = render(
      <GlobalStateProvider>
        <ApplicationThemeProvider theme={defaultTheme}>
          <Peril index={0} onClick={handleClick} peril={peril} />
        </ApplicationThemeProvider>
      </GlobalStateProvider>
    );
    const button = getByTestId('Peril');

    fireEvent.click(button);
    expect(handleClick).toHaveBeenCalledTimes(1);

    fireEvent.click(button);
    expect(handleClick).toHaveBeenCalledTimes(2);
  });
});
