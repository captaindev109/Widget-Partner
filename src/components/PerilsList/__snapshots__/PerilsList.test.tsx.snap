// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Peril snapshot test Snapshot test should pass 1`] = `
<ul
  className="List__StyledList-sc-1b1or9q-0 cjSnKm"
  direction="row"
  display="flex"
  wrap="wrap"
>
  <li
    className="List__StyledListItem-sc-1b1or9q-1 PerilsList__PerilsListItem-sc-9z80pk-0 itixlA"
  >
    <div
      className="Card-sc-fb64gk-0 PerilsList__StyledPeril-sc-9z80pk-1 iJgSZC lgGhXf"
      data-testid="Peril"
      onClick={[Function]}
    >
      <img
        alt="Eldsvåda"
        className="Image__StyledImage-sc-34jhra-0 kHOSNn"
        height="20px"
        src="https://promise.hedvig.com/media/fire_97796e0790.svg"
        width="20px"
      />
      <p
        className="Text__StyledText-sc-1jfsuh1-0 iWpXoI PerilsList__PerilText-sc-9z80pk-2 cnFFLb"
        color="gray900"
      >
        Eldsvåda
      </p>
    </div>
  </li>
  <li
    className="List__StyledListItem-sc-1b1or9q-1 PerilsList__PerilsListItem-sc-9z80pk-0 itixlA"
  >
    <div
      className="Card-sc-fb64gk-0 PerilsList__StyledPeril-sc-9z80pk-1 iJgSZC ksaMNF"
      data-testid="Peril"
      onClick={[Function]}
    >
      <img
        alt="Vattenläcka"
        className="Image__StyledImage-sc-34jhra-0 kHOSNn"
        height="20px"
        src="https://promise.hedvig.com/media/water_damage_e25b83cd0b.svg"
        width="20px"
      />
      <p
        className="Text__StyledText-sc-1jfsuh1-0 iWpXoI PerilsList__PerilText-sc-9z80pk-2 cnFFLb"
        color="gray900"
      >
        Vattenläcka
      </p>
    </div>
  </li>
  <li
    className="List__StyledListItem-sc-1b1or9q-1 PerilsList__PerilsListItem-sc-9z80pk-0 itixlA"
  >
    <div
      className="Card-sc-fb64gk-0 PerilsList__StyledPeril-sc-9z80pk-1 iJgSZC ikvspK"
      data-testid="Peril"
      onClick={[Function]}
    >
      <img
        alt="Inbrott"
        className="Image__StyledImage-sc-34jhra-0 kHOSNn"
        height="20px"
        src="https://promise.hedvig.com/media/burglary_946096433f.svg"
        width="20px"
      />
      <p
        className="Text__StyledText-sc-1jfsuh1-0 iWpXoI PerilsList__PerilText-sc-9z80pk-2 cnFFLb"
        color="gray900"
      >
        Inbrott
      </p>
    </div>
  </li>
  <li
    className="List__StyledListItem-sc-1b1or9q-1 PerilsList__PerilsListItem-sc-9z80pk-0 itixlA"
  >
    <div
      className="Card-sc-fb64gk-0 PerilsList__StyledPeril-sc-9z80pk-1 iJgSZC TYTmY"
      data-testid="Peril"
      onClick={[Function]}
    >
      <img
        alt="Stöld"
        className="Image__StyledImage-sc-34jhra-0 kHOSNn"
        height="20px"
        src="https://promise.hedvig.com/media/theft_701fa78317.svg"
        width="20px"
      />
      <p
        className="Text__StyledText-sc-1jfsuh1-0 iWpXoI PerilsList__PerilText-sc-9z80pk-2 cnFFLb"
        color="gray900"
      >
        Stöld
      </p>
    </div>
  </li>
  <li
    className="List__StyledListItem-sc-1b1or9q-1 PerilsList__PerilsListItem-sc-9z80pk-0 itixlA"
  >
    <div
      className="Card-sc-fb64gk-0 PerilsList__StyledPeril-sc-9z80pk-1 iJgSZC lgGhXf"
      data-testid="Peril"
      onClick={[Function]}
    >
      <img
        alt="Skadegörelse"
        className="Image__StyledImage-sc-34jhra-0 kHOSNn"
        height="20px"
        src="https://promise.hedvig.com/media/criminal_damage_9f725dc7cf.svg"
        width="20px"
      />
      <p
        className="Text__StyledText-sc-1jfsuh1-0 iWpXoI PerilsList__PerilText-sc-9z80pk-2 cnFFLb"
        color="gray900"
      >
        Skadegörelse
      </p>
    </div>
  </li>
  <li
    className="List__StyledListItem-sc-1b1or9q-1 PerilsList__PerilsListItem-sc-9z80pk-0 itixlA"
  >
    <div
      className="Card-sc-fb64gk-0 PerilsList__StyledPeril-sc-9z80pk-1 iJgSZC ksaMNF"
      data-testid="Peril"
      onClick={[Function]}
    >
      <img
        alt="Ansvarsskydd"
        className="Image__StyledImage-sc-34jhra-0 kHOSNn"
        height="20px"
        src="https://promise.hedvig.com/media/liability_c331907aff.svg"
        width="20px"
      />
      <p
        className="Text__StyledText-sc-1jfsuh1-0 iWpXoI PerilsList__PerilText-sc-9z80pk-2 cnFFLb"
        color="gray900"
      >
        Ansvarsskydd
      </p>
    </div>
  </li>
  <li
    className="List__StyledListItem-sc-1b1or9q-1 PerilsList__PerilsListItem-sc-9z80pk-0 itixlA"
  >
    <div
      className="Card-sc-fb64gk-0 PerilsList__StyledPeril-sc-9z80pk-1 iJgSZC ikvspK"
      data-testid="Peril"
      onClick={[Function]}
    >
      <img
        alt="Rättsskydd"
        className="Image__StyledImage-sc-34jhra-0 kHOSNn"
        height="20px"
        src="https://promise.hedvig.com/media/legal_protection_f6e65e1b13.svg"
        width="20px"
      />
      <p
        className="Text__StyledText-sc-1jfsuh1-0 iWpXoI PerilsList__PerilText-sc-9z80pk-2 cnFFLb"
        color="gray900"
      >
        Rättsskydd
      </p>
    </div>
  </li>
  <li
    className="List__StyledListItem-sc-1b1or9q-1 PerilsList__PerilsListItem-sc-9z80pk-0 itixlA"
  >
    <div
      className="Card-sc-fb64gk-0 PerilsList__StyledPeril-sc-9z80pk-1 iJgSZC TYTmY"
      data-testid="Peril"
      onClick={[Function]}
    >
      <img
        alt="Resetrubbel"
        className="Image__StyledImage-sc-34jhra-0 kHOSNn"
        height="20px"
        src="https://promise.hedvig.com/media/travel_insurance_ed0fa95fdf.svg"
        width="20px"
      />
      <p
        className="Text__StyledText-sc-1jfsuh1-0 iWpXoI PerilsList__PerilText-sc-9z80pk-2 cnFFLb"
        color="gray900"
      >
        Resetrubbel
      </p>
    </div>
  </li>
  <li
    className="List__StyledListItem-sc-1b1or9q-1 PerilsList__PerilsListItem-sc-9z80pk-0 itixlA"
  >
    <div
      className="Card-sc-fb64gk-0 PerilsList__StyledPeril-sc-9z80pk-1 iJgSZC lgGhXf"
      data-testid="Peril"
      onClick={[Function]}
    >
      <img
        alt="Överfall"
        className="Image__StyledImage-sc-34jhra-0 kHOSNn"
        height="20px"
        src="https://promise.hedvig.com/media/assault_d371c026ac.svg"
        width="20px"
      />
      <p
        className="Text__StyledText-sc-1jfsuh1-0 iWpXoI PerilsList__PerilText-sc-9z80pk-2 cnFFLb"
        color="gray900"
      >
        Överfall
      </p>
    </div>
  </li>
  <li
    className="List__StyledListItem-sc-1b1or9q-1 PerilsList__PerilsListItem-sc-9z80pk-0 itixlA"
  >
    <div
      className="Card-sc-fb64gk-0 PerilsList__StyledPeril-sc-9z80pk-1 iJgSZC ksaMNF"
      data-testid="Peril"
      onClick={[Function]}
    >
      <img
        alt="Sjuk på resa"
        className="Image__StyledImage-sc-34jhra-0 kHOSNn"
        height="20px"
        src="https://promise.hedvig.com/media/sick_on_holiday_17b3f1899c.svg"
        width="20px"
      />
      <p
        className="Text__StyledText-sc-1jfsuh1-0 iWpXoI PerilsList__PerilText-sc-9z80pk-2 cnFFLb"
        color="gray900"
      >
        Sjuk på resa
      </p>
    </div>
  </li>
  <li
    className="List__StyledListItem-sc-1b1or9q-1 PerilsList__PerilsListItem-sc-9z80pk-0 itixlA"
  >
    <div
      className="Card-sc-fb64gk-0 PerilsList__StyledPeril-sc-9z80pk-1 iJgSZC ikvspK"
      data-testid="Peril"
      onClick={[Function]}
    >
      <img
        alt="Vitvaror"
        className="Image__StyledImage-sc-34jhra-0 kHOSNn"
        height="20px"
        src="https://promise.hedvig.com/media/appliance_damage_2a23c61a09.svg"
        width="20px"
      />
      <p
        className="Text__StyledText-sc-1jfsuh1-0 iWpXoI PerilsList__PerilText-sc-9z80pk-2 cnFFLb"
        color="gray900"
      >
        Vitvaror
      </p>
    </div>
  </li>
  <li
    className="List__StyledListItem-sc-1b1or9q-1 PerilsList__PerilsListItem-sc-9z80pk-0 itixlA"
  >
    <div
      className="Card-sc-fb64gk-0 PerilsList__StyledPeril-sc-9z80pk-1 iJgSZC TYTmY"
      data-testid="Peril"
      onClick={[Function]}
    >
      <img
        alt="Drulle"
        className="Image__StyledImage-sc-34jhra-0 kHOSNn"
        height="20px"
        src="https://promise.hedvig.com/media/all_risk_eba77dfb03.svg"
        width="20px"
      />
      <p
        className="Text__StyledText-sc-1jfsuh1-0 iWpXoI PerilsList__PerilText-sc-9z80pk-2 cnFFLb"
        color="gray900"
      >
        Drulle
      </p>
    </div>
  </li>
  <li
    className="List__StyledListItem-sc-1b1or9q-1 PerilsList__PerilsListItem-sc-9z80pk-0 itixlA"
  >
    <div
      className="Card-sc-fb64gk-0 PerilsList__StyledPeril-sc-9z80pk-1 iJgSZC lgGhXf"
      data-testid="Peril"
      onClick={[Function]}
    >
      <img
        alt="Bostadsrättstillägg"
        className="Image__StyledImage-sc-34jhra-0 kHOSNn"
        height="20px"
        src="https://promise.hedvig.com/media/brf_additional_51b2fb160a.svg"
        width="20px"
      />
      <p
        className="Text__StyledText-sc-1jfsuh1-0 iWpXoI PerilsList__PerilText-sc-9z80pk-2 cnFFLb"
        color="gray900"
      >
        Bostadsrättstillägg
      </p>
    </div>
  </li>
</ul>
`;
